Source/webserver/src/main/java/klingon/webserver/WebserverApplication.java

```diff
@@ -3,22 +3,135 @@
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+import org.springframework.stereotype.Service;
 
 import java.util.List;
 
+/**
+ * This class is a initializer to boot up the applications backend
+ * and handles different execution of the built in database
+ *
+ *  * @author Anthon Lenander, Phong Nguyen
+ *  * @version 2021-09-21
+ *
+ */
+
+@Service
 @SpringBootApplication
 public class WebserverApplication
 {
+	private static BicycleStationRepository bicycleStationRepository;
+	private static PumpStationRepository pumpStationRepository;
+
 	public static void main(String[] args)
 	{
 		ConfigurableApplicationContext context = SpringApplication.run(WebserverApplication.class, args);
-    	BicycleStationRepository bicycleStationRepository = context.getBean(BicycleStationRepository.class);
-		PumpStationRepository pumpStationRepository = context.getBean(PumpStationRepository.class);
+    	bicycleStationRepository = context.getBean(BicycleStationRepository.class);
+		pumpStationRepository = context.getBean(PumpStationRepository.class);
+		context.start();
+	}
+
+	/**
+	 * Get method that returns the repository of all bicycle stations
+	 *
+	 * @return a BicycleStationRepository
+	 */
+	public static BicycleStationRepository getBicycleStationRepository()
+	{
+		return bicycleStationRepository;
+	}
+
+	/**
+	 * Get method that returns the repository of all pump stations
+	 *
+	 * @return a PumpStationRepository
+	 */
+	public static PumpStationRepository getPumpStationRepository()
+	{
+		return pumpStationRepository;
+	}
+
+	// The spring cron expression should be formatted as follows:
+	// seconds minutes hours day_of_month month day(s)_of_week.
+
+	@Scheduled(cron = "0 */5 * * * *")
+	protected void updateBicycleStations()
+	{
+		populateBicycleStations();
+	}
+	
+	@Scheduled(cron = "0 0 8 * * 1")
+	protected void updatePumpStations()
+	{
+		populatePumpStations();
+	}
+
+	private void populateBicycleStations()
+	{
+		List<BicycleStation> bicycleStations = APIDataHandler.getBicycleStationData();
+		bicycleStationRepository.saveAll(bicycleStations);
+	}
 
-    	List<BicycleStation> bicycleStations = APIDataHandler.getBicycleStationData();
-    	List<PumpStation> allPumpStations = APIDataHandler.getAllPumpStations();
-    
-    	bicycleStationRepository.saveAll(bicycleStations);
+	private void populatePumpStations()
+	{
+		List<PumpStation> allPumpStations = APIDataHandler.getAllPumpStations();
 		pumpStationRepository.saveAll(allPumpStations);
 	}
+
+	private void initDatabase()
+	{
+		bicycleStationRepository.deleteAll();
+		pumpStationRepository.deleteAll();
+
+		populateBicycleStations();
+		populatePumpStations();
+	}
+
+	@EventListener(ContextStartedEvent.class)
+	protected void onApplicationStartup()
+	{
+		initDatabase();
+
+		updateBicycleStations();
+		updatePumpStations();
+	}
+}
+
+@Configuration
+@EnableAsync
+@EnableScheduling
+class SchedulingConfiguration implements AsyncConfigurer
```

Comment:
Preferable, one top-level class per file. I think it would be better to move SchedulingConfiguration to a new file.

Reply:
I was unsure if moving this piece of code to a separate file would impact anything during runtime, but I have now moved it into SchedulingConfiguration.java, tested it, and it seems to work just fine.